testsuite/MDAnalysisTests/coordinates/test_h5md.py

```diff
@@ -493,9 +358,325 @@ def test_no_units(h5md_file, ref, tmpdir):
+        (None, None, 'c', None),
+        (None, None, None, 'HUGE FORCE',),
+        (None, 'lightyear', None, None)))
+    def test_write_bad_units(self, universe, outfile, Writer,
+                             timeunit, lengthunit,
+                             velocityunit, forceunit):
+        with pytest.raises(RuntimeError):
+            with Writer(outfile,
+                        universe.atoms.n_atoms,
+                        velocities=True,
+                        forces=True,
+                        lengthunit=lengthunit,
+                        velocityunit=velocityunit,
+                        forceunit=forceunit,
+                        timeunit=timeunit) as W:
+                for ts in universe.trajectory:
+                    W.write(universe)
+
+    @pytest.mark.parametrize('convert_units, units', (
+        (True, ['ps', 'nm', 'nm/ps', 'kJ/(mol*nm)']),
+        (False, ['ps', 'Angstrom', 'Angstrom/ps', 'kJ/(mol*Angstrom)'])))
+    def test_convert_units(self, universe, outfile, Writer,
+                           convert_units, units):
+        with Writer(outfile,
+                    universe.atoms.n_atoms,
+                    velocities=True,
+                    forces=True,
+                    convert_units=convert_units) as W:
+            for ts in universe.trajectory:
+                W.write(universe)
+
+        uw = mda.Universe(TPR_xvf, outfile)
+        uw_units = uw.trajectory.units.values()
+        for u1, u2 in zip(units, uw_units):
+            assert_equal(u1, u2)
+
+    #@pytest.mark.parametrize('chunks', (None, (1, 1000, 3)))
+    def test_write_chunks(self, universe, outfile, Writer):
+        chunks = (1, 1000, 3)
+        with Writer(outfile,
+                    universe.atoms.n_atoms,
+                    chunks=chunks) as W:
+            for ts in universe.trajectory:
+                W.write(universe)
+
+        uw = mda.Universe(TPR_xvf, outfile)
+        dset = uw.trajectory._particle_group['position/value']
+        assert_equal(dset.chunks, chunks)
+
+    @pytest.mark.parametrize('filter, opts', (('gzip', 1),
+                                              ('gzip', 9),
+                                              ('lzf', None)))
+    def test_write_with_compression(self, universe,
+                                    outfile, Writer,
+                                    filter, opts):
+        with Writer(outfile,
+                    universe.atoms.n_atoms,
+                    compression=filter,
+                    compression_opts=opts) as W:
+            for ts in universe.trajectory:
+                W.write(universe)
+
+        uw = mda.Universe(TPR_xvf, outfile)
+        dset = uw.trajectory._particle_group['position/value']
+        assert_equal(dset.compression, filter)
+        assert_equal(dset.compression_opts, opts)
+
+    @pytest.mark.parametrize('driver, comm', (('core', None),
+                                              ('mpio', 'comm')))
```

Comment:
The try/except inside to distinguish these two cases looks unnecessary complicated and introduces additional uncertainty (the blanket `except`). Why not make two different tests?

Reply:
two separate tests makes more sense